src/detect-engine-threshold.c

```diff
@@ -713,6 +672,32 @@ void ThresholdHashInit(DetectEngineCtx *de_ctx)
     }
 }
 
+/**
+ * \brief Realloc threshold context hash tables
+ *
+ * \param de_ctx Detection Context
+ */
+void ThresholdHashRealloc(DetectEngineCtx *de_ctx)
+{
+    /* Return if we are already big enough */
+    uint32_t num = de_ctx->signum + 1;
+    if (num <= de_ctx->ths_ctx.th_size)
+        return;
+
+    void *ptmp = SCRealloc(de_ctx->ths_ctx.th_entry, num * sizeof(DetectThresholdEntry *));
+    if (ptmp == NULL) {
+        SCLogError(SC_ERR_MEM_ALLOC, "Error allocating memory for rule thresholds"
+                " (tried to allocate %"PRIu32" th_entrys for rule tracking)", num);
+        exit(EXIT_FAILURE);
```

Comment:
This is a change in behavior -- previously, an error message was logged with incomplete cleanup.

@victorjulien Should this be a warning instead?  When `realloc` fails, the original pointer is left unmodified.

Reply:
Oh, true. I must have been looking at the initialization routines when I added this. I can remove it, and have the function warn instead without changing the table.

Reply:
Yeah lets do the warn thing, even if that also isn't ideal. It's at least not changing the behaviour.

Reply:
Done in #4747. The previous behaviour was to free the existing entry on allocation failure, so that's what we do now.